galsim_extra/mixed_scene.py

```diff
@@ -56,10 +58,34 @@ def setup(self, config, base, xsize, ysize, ignore, logger):
         # Add objects field to the ignore list
         # Also ignore magnify and shear, which we allow here for convenience to act on whichever
         # object ends up being chosen.
-        ignore = ignore + ['objects', 'magnify', 'shear', 'obj_type']
+        ignore = ignore + ['objects', 'magnify', 'shear', 'obj_type', 'shear_scene']
+
+        stamp_xsize, stamp_ysize, image_pos, world_pos = super(MixedSceneBuilder, self).setup(config,base,xsize,ysize,ignore,logger)
+        
+        if 'shear_scene' in config:
+            shear_scene = galsim.config.ParseValue(config, 'shear_scene', base, bool)[0]
+        else:
+            shear_scene = False
+        
+        # option to shear the full scene.
+        if shear_scene:       
+            shear = galsim.config.ParseValue(config, 'shear', base, galsim.Shear)[0]
+            # Find the center (tangent point) of the scene in RA, DEC. 
+            scene_center = base['world_center']
+            wcs = base['wcs']
+            if wcs.isCelestial:
+                u, v = scene_center.project(world_pos, projection='gnomonic')
+                pos = galsim.PositionD(u.rad, v.rad)
+                sheared_pos = pos.shear(shear)
+                u2 = sheared_pos.x * coord.radians
+                v2 = sheared_pos.y * coord.radians
+                world_pos = scene_center.deproject(u2, v2, projection='gnomonic')
```

Comment:
@rmjarvis Is always using a gnomic projection ok here? Should we maybe match the projection assumed by the WCS?

Reply:
Yeah, that should be ok.  Definitely doesn't matter what the wcs is using, since this projection has nothing to do with that.  You're just trying to get a flat sky approximation around the image center.  Honestly, if I had to guess, I think `postel` is probably the best choice, since it's an equidistant projection.  But they're all probably close enough to equivalent that it doesn't matter.

Reply:
🤔